concrete/src/Job/Job.php

```diff
@@ -207,219 +254,86 @@ public function markCompleted($resultCode = 0, $resultMsg = false)
-                        $jobObjs[$jHandle] = Core::make($className);
-                        $jobObjs[$jHandle]->jHandle = $jHandle;
-                    }
-                    closedir($dh);
-                }
-            }
-        }
-
-        return $jobObjs;
-    }
-
-    // Running Jobs
-    // ==============
-
-    /*
-    public static function runAllJobs()
-    {
-        //loop through all installed jobs
-        $jobs = Job::getList();
-        foreach ($jobs as $j) {
-            $j->executeJob();
-        }
-    }
-    */
-
+    /**
+     * @return JobResult
+     */
     public function executeJob()
     {
+        $error = '';
         $this->markStarted();
+
         try {
             $resultMsg = $this->run();
-            if (strlen($resultMsg) == 0) {
+
+            if (strlen($resultMsg) === 0) {
                 $resultMsg = t('The Job was run successfully.');
             }
-        } catch (\Exception $e) {
+        } catch (Exception $e) {
             $resultMsg = $e->getMessage();
             $error = static::JOB_ERROR_EXCEPTION_GENERAL;
         }
 
-        $je = new Event($this);
-        Events::dispatch('on_job_execute', $je);
-
         $obj = $this->markCompleted($error, $resultMsg);
 
         return $obj;
     }
 
-    public function setJobStatus($jStatus = 'ENABLED')
+    /**
+     * @param string $jStatus
+     *
+     * @return Job $this
+     */
+    public function setJobStatus($jStatus = self::JOB_STATUS_ENABLED)
     {
-        $db = Loader::db();
         if (!in_array($jStatus, $this->availableJStatus)) {
-            $jStatus = 'ENABLED';
+            $jStatus = self::JOB_STATUS_ENABLED;
         }
-        $rs = $db->query("UPDATE Jobs SET jStatus=? WHERE jHandle=?", array($jStatus, $this->jHandle));
-    }
 
-    public static function installByHandle($jHandle = '')
-    {
-        $availableJobs = static::getAvailableList();
-        foreach ($availableJobs as $availableJobHandle => $availableJobObj) {
-            if ($availableJobHandle != $jHandle) {
-                continue;
-            }
-            $availableJobObj->install();
-        }
-    }
+        $this->jStatus = $jStatus;
 
-    public static function getListByPackage($pkg)
-    {
-        $db = Loader::db();
-        $list = array();
-        $r = $db->Execute('select jHandle from Jobs where pkgID = ? order by jHandle asc', array($pkg->getPackageID()));
-        while ($row = $r->FetchRow()) {
-            $list[] = static::getJobObjByHandle($row['jHandle']);
-        }
-        $r->Close();
+        $db = static::$app['database'];
+        $db->executeQuery("UPDATE Jobs SET jStatus=? WHERE jHandle=?", [
+            $this->jStatus,
+            $this->jHandle,
+        ]);
 
-        return $list;
+        return $this;
     }
 
-    public static function installByPackage($jHandle, $pkg)
+    /**
+     * @return Job $this
+     */
+    public function install()
```

Comment:
Am I supposed to install using `$package->install()` or `$packageService->install($package)`?